stl/inc/format

```diff
@@ -441,20 +441,35 @@ _NODISCARD constexpr const _CharT* _Parse_arg_id(
     // This is where we would parse named arg ids if std::format were to support them.
     _THROW(format_error("Invalid format string."));
 }
-
-_NODISCARD constexpr bool _Is_execution_charset_utf8() {
-#pragma warning(push)
-#pragma warning(disable : 4309) // 'initializing' : truncation of constant value
-#pragma warning(disable : 4566) // character represented by universal-character-name '\u4E00' cannot be represented in
-                                // the current code page
-#pragma warning(disable : 6201) // Index '2' is out of valid index range '0' to '1' for possibly stack allocated buffer
-                                // '_Test_char'
-#pragma warning(disable : 6239) // (<non-zero constant> && <expression>) always evaluates to the result of <expression>.
-                                // Did you intend to use the bitwise-and operator?
-    constexpr char _Test_char[] = "\u4e00";
-    return sizeof(_Test_char) == 4 && _Test_char[0] == '\xe4' && _Test_char[1] == '\xb8' && _Test_char[2] == '\x80';
-#pragma warning(pop)
+#ifdef __clang__ // TRANSITION: LLVM-52549 (clang-cl 14 will define _MSVC_EXECUTION_CHARACTER_SET)
+_NODISCARD constexpr bool _Is_execution_charset_self_synchronizing() {
+    return string_view{__clang_literal_encoding__} == "UTF-8";
+}
+#else // ^^^ clang / msvc vvv
+_NODISCARD constexpr bool _Is_execution_charset_self_synchronizing() {
+
+    // charset codes (as reported by _MSVC_EXECUTION_CHARACTER_SET) that are
+    // self-synchronizing (and "narrow" so 1200 (utf-16) and 1201 (unicodeFFFE) are not listed)
+    // we care about this because if a charset is self-synchronizing then we can search through it
+    // for formatting control characters _without_ generally decoding the format string.
+    switch (_MSVC_EXECUTION_CHARACTER_SET) {

```

Comment:
These names *exactly* match https://docs.microsoft.com/en-us/windows/win32/intl/code-page-identifiers which is excellent, thank you. I additionally recommend citing this documentation so that this remains consistent if the list needs to be changed in the future.
```suggestion
    switch (_MSVC_EXECUTION_CHARACTER_SET) {
    // See: https://docs.microsoft.com/en-us/windows/win32/intl/code-page-identifiers
```